### PR TITLE
Expand root-level synthetic IEnumerable values without intermediate "Results View"

### PR DESCRIPTION
**Pull request by cston**
Expand root-level synthetic IEnumerable values without intermediate "Results View"

Removing the intermediate "Results View" row for synthetic values allows LINQ queries to be expanded automatically in the Immediate Window.

### CODE REVIEW

**Comment by cston** [comment]
@amcasey, @KevinH-MS, @pnelsonmsft, @ManishJayaswal

File: src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs [hunk 1]
```diff
@@ -59,30 +59,89 @@ internal static ResultsViewExpansion CreateExpansion(DkmInspectionContext inspec
...
+            DkmClrType declaredType,
+            DkmClrValue value,
+            Formatter formatter)
+        {
+            if ((value.ValueFlags & DkmClrValueFlags.Synthetic) == 0)
+            {
+                return null;
+            }
+
+            if (!IsEnumerableCandidate(value))
```
Comment (KevinH-MS) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs]: Can this "if" be moved into GetEnumerableType (and removed here and below)?

File: src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs [hunk 2]
```diff
@@ -176,5 +232,56 @@ private EvalResultDataItem CreateResultsViewRow(DkmInspectionContext inspectionC
                 editableValue: null,
                 inspectionContext: inspectionContext);
         }
+
+        private EvalResultDataItem CreateResultsViewRow(DkmInspectionContext inspectionContext, string name, Type declaredType, DkmClrValue value, Formatter formatter)
```
Comment (KevinH-MS) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs]: Could we just call this overload of CreateResultsViewRow from CreateResultsOnlyRow (and then remove anything that's no longer needed from the other overload of CreateResultsViewRow)?  It's always seemed odd to me that if you do "e, results", we say the Value is "Expanding the Results View will enumerate the IEnumerable".  It seems like this new behavior would be better in that case as well.

**Comment by KevinH-MS** [comment]
LGTM :shipit:

File: src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs (hunk 1 above)
Comment (cston) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs]: Done.

File: src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs (hunk 2 above)
Comment (cston) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs]: That makes sense. I'll investigate that change separately from the rest of this change though.

Comment (KevinH-MS) [inline src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/ResultsViewExpansion.cs]: Sounds good.

**Pull request merged**